app/actions/handlers.py

```diff
@@ -15,11 +14,16 @@
 state_manager = IntegrationStateManager()
 
 
-def extract_gmt_offsets(transmissions):
-    accumulator = {}
-    for item in transmissions:
-        accumulator.setdefault(item.collar_serial_num, item.gmt_offset)
-    return accumulator
+def extract_gmt_offsets(transmissions, integration_id):
+    if transmissions:
+        accumulator = {}
+        for item in transmissions:
+            accumulator.setdefault(item.collar_serial_num, item.gmt_offset)
+        return accumulator
```

Comment:
Since this is a dictionary, there won't be duplicate keys and the dict size won't grow. Ultimately, if there are several records for the same key (serial num) then the value will be updated several times and only the last one will remain. But if you only want to save the first offset you see for a collar, and you want to avoid those extra updates then makes sense to use `setdefault` as you did.